### PR TITLE
 Adding interactive to do list

### PR DESCRIPTION
**Pull request by DamenaM**
###  PR for an interactive to-do list

- [ ] Add event listener to the checkbox
- [ ] implement a function for the Clear all completed task 
- [ ] Store the updated array of items in local storage
- [ ] Update to-do list  items

### CODE REVIEW

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @DamenaM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive PR ✔️ 
- No linter errors ✔️
- Correct Github flow ✔️ 
- Array.filter() method is implemented 💯 
- The updated array of items is stored in local storage 💯  

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

I will use the summary to add my suggestion because it is not possible to use inline comments:

**General requirement:**

- [ ] Your README should look professional id you have empty sections it doesn't look professional so please 🙏🏽 complete the Acknowledgment section, it is mandatory:

image

**Project requirements:**

- [ ] It could be better to use two files ✌🏽 at least in the modules folder to be able to import functions between modules and avoid extended pages of code I could recommend creating a file that contains the classes and another that contain the functions (add, edit, delete, clear all)

- **[OPTIONAL]** Kindly do not disable the liters on code lines it helps you to have clean code, you have this warning because you declare a function with an argument (index) that you didn't use on it. Please 🙏🏽 refactor your code to correct the liter errors it will help you to have clean code.

image

**Functionalities:**

- [ ] When I edited a task 📑 🖊️, the rest of the activities disappeared. Please correct this behavior for a better user experience.

- [ ] Even the status of the checkbox ☑️ updated well, please refactor your code and add the event listener (change), this is a project requirement, you need to complete them all to approve.

> Add event listener to the checkbox (change).

**Styles:**

For a better user interface:

- **[OPTIONAL]** It could be better to ➕ add padding and a margin between the add form and the list of activities.

- **[OPTIONAL]** I think it will look nicer 👍🏽 if the buttons edit and delete would have the same size and be aligned.

- **[OPTIONAL]** Please 🙏🏽 align the center of the Clear all completed button to look similar to the original template.

| Original design | Your design |
|---|---|
| image | image |



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Module/task.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    newEditTask(index) {  // eslint-disable-line
+    // eslint-disable-next-line
+      const toBeEdited = event.target.parentElement.parentElement.firstChild.nextElementSibling.lastChild.previousElementSibling;
+    toBeEdited.contentEditable = 'true';
+    toBeEdited.style.border = '1px solid black';
+
+      if (event.target.classList.contains('fa-solid')) {// eslint-disable-line
+
+        event.target.parentElement.parentElement.style.backgroundColor = ' #f6ee78';// eslint-disable-line
+    }
```
Comment (mikethreels) [inline src/Module/task.js:34]: - [ ] when editing a task or reloading a page the status of all the tasks is reset, please fix this before requesting another review.

File: src/Module/clearall [hunk 2]
```diff
@@ -0,0 +1 @@
+export default (tasks) => tasks.filter((task) => !task.completed);
```
Comment (mikethreels) [inline src/Module/clearall:1]: - [ ] when marking several tasks as completed and trying to remove the completed tasks this also removes tasks which weren't marked as completed, please fix this.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi MIhertab, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @DamenaM 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Github flow was used correctly 🚀 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**